### ISSUE TITLE
Build a multipart request with attach() and an XML payload

### ISSUE THREAD

**Issue by flandrade** [visionmedia/superagent#1358]
We are able to use an XML payload with the following configuration: 

```ts
.post("/api/...")
  .type("text/xml")
  .send(xmlPayload)
```

However, if you need a multipart request, it's not possible to use `send` anymore. According to the documentation: 
> When you use .field() or .attach() you can't use .send() and you must not set Content-Type (the correct type will be set for you).

Is there a way to build a multipart request with `.attach()` and an XML payload? I'm unable to do it with `.field()`.

**Comment by kornelski** [comment]
Yes, `.attach("multipart field name", new Blob([xmlPayload], {type:"text/xml"})`

**Comment by flandrade** [comment]
@kornelski Thanks, this works if you attach the XML, but I'm trying to reproduce "SwA" or SOAP with Attachments, where you a body has a request payload in XML and an attached file. Is this possible?

**Comment by kornelski** [comment]
If it's the Multipart/Related format you're after, then it's not possible unless you build the whole MIME message from scratch the hard way.
